truss/templates/server/common/truss_server.py

```diff
@@ -278,6 +279,28 @@ def create_application(self):
             },
         )
 
+        async def intercept_request_id(request, call_next):
+            request_id = request.headers.get("x-baseten-request-id", None)
+
+            with loguru_logger.contextualize(
+                request_id=request_id, lifecycle=Lifecycle.REQUEST
+            ):
+                response = await call_next(request)
+
+                # don't log health checks
+                if not request.state.is_health_check:
+                    loguru_logger.info(
+                        f"{request.method} {request.url.path} {response.status_code}"
+                    )
+                return response
+
+        async def access_logging(request, call_next):
+            request.state.is_health_check = (
```

Comment:
could we move the actual access logging here

Reply:
Done!